### PR TITLE
feat: Request auth token for downloading Hub plugins

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Part of https://github.com/cloudquery/cloudquery-issues/issues/1069 (internal issue)

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt ./...` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: managedplugin/download.go [hunk 1]
```diff
@@ -128,6 +128,9 @@ type HubDownloadOptions struct {
 }
 
 func DownloadPluginFromHub(ctx context.Context, ops HubDownloadOptions) error {
+	if ops.AuthToken == "" {
+		return fmt.Errorf("an authentication token is required to download plugins from the CloudQuery registry. Try logging in via `cloudquery login`")
```
Comment (disq) [inline managedplugin/download.go:132]: I would make this an exported error (e.g. `ErrRequiresLogin`) and remove the "Try logging in" part. Which can be added via the CLI later, after checking with `errors.Is`?

Comment (erezrokah) [reply managedplugin/download.go]: Good suggestion, extracted in https://github.com/cloudquery/plugin-pb-go/pull/200/commits/42aed399a8cc95407cc9a889b8d8fed45897f0e9

Comment (erezrokah) [reply managedplugin/download.go]: We probably would want to add more errors from https://github.com/cloudquery/plugin-pb-go/blob/f5e58744d876169e3761755ca38ab0462f0228a6/managedplugin/download.go#L144 in a separate PR

**Comment by erezrokah** [comment]
Should be released as a CLI major version

**Comment by erezrokah** [comment]
Closing as stale. I don't think we need this and we can let the CLI fail with the backend error

**Pull request closed without merging**